master/buildbot/util/croniter.py

```diff
@@ -67,18 +97,58 @@ def __init__(self, expr_format, start_time=time()):
             raise ValueError(self.bad_length)
 
         expanded = []
+        nth_weekday_of_month = {}
 
         for i, expr in enumerate(self.exprs):
+            expr = expr.lower()
             e_list = expr.split(',')
             res = []
 
             while e_list:
                 e = e_list.pop()
+                nth = None
+                special_dow_rem = special_dow_re.match(str(e))
+
+                # Handle special case in the day of week expression: 2#3, l3
+                if i == 4 and special_dow_rem:
```

Comment:
Let's not use `4` as a special constant, but introduce names for them, e.g. `cron_min_index`, `cron_day_of_month_index`, etc.